### PR TITLE
[BE] Use unwrap_sym_or_const helper function

### PR DESCRIPTION
**Pull request by yanboliang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #137077
* #137075

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @rec

### CODE REVIEW

File: torch/_dynamo/utils.py [hunk 1]
```diff
@@ -1373,12 +1373,11 @@ def is_safe_constant(v):
 
 
 def unwrap_sym_or_const(arg):
-    from .variables import ConstantVariable, SymNodeVariable
+    from .variables import SymNodeVariable
 
     if isinstance(arg, SymNodeVariable):
         return arg.sym_num
     else:
-        assert isinstance(arg, ConstantVariable)
```
Comment (Skylion007) [inline torch/_dynamo/utils.py:1381]: Should we just add the other isinstance arg here?

File: test/functorch/test_vmap.py [hunk 2]
```diff
@@ -3366,6 +3366,7 @@ def f(x):
             )(x)
             self.assertEqual(output, expected)
 
+    @skipIfTorchDynamo()
```
Comment (zou3519) [inline test/functorch/test_vmap.py:3369]: these used to pass, what's going on?

Comment (yanboliang) [reply test/functorch/test_vmap.py:3369]: The eager backend issue we discussed at HOP meeting, but I think I'll gave up this PR stack since it messed up.

**Comment by yanboliang** [comment]
closing this in favor of #137163

**Pull request closed without merging**